src/content/en/showcase/2016/service-worker-perf.md

```diff
@@ -0,0 +1,679 @@
+ga('send', 'pageview');
+
+// Sends an event with the time to first paint data.
+sendTimeToFirstPaint();
+</pre>
+
+Note that depending on when the above code runs, pixel may or may not have already been painted to the screen. To ensure we always run this code after the first paint occurs, we postponed the call to `sendTimeToFirstPaint()` until after the `load` event. In fact, we decided to postpone sending all analytics data until after the page was loaded to ensure those requests wouldn't compete with the loading of other resources.
+
+<pre class="prettyprint">
+// Creates the tracker object.
+ga('create', 'UA-XXXXX-Y', 'auto');
+
+// Postpones sending any hits until after the page has fully loaded.
+// This prevents analytics requests from delaying the loading of the page.
+<strong>window.addEventListener('load', function() {
+  // Sends a pageview for the initial pageload.
+  ga('send', 'pageview');
+
+  // Sends an event with the time to first paint data.
+  sendTimeToFirstPaint();
+});</strong>
+</pre>
+
+The code above reports `firstpaint` times to Google Analytics, but that's only half the story. We still needed to track the service worker status; otherwise we wouldn't be able to compare the first paint times of a service worker-controlled page and a non-controlled page.
+
+### Determining service worker status
+
+To determine the current status of the service worker, we created a utility function that returns  one of three values:
+
+- **controlled**: a service worker is controlling the page. In the case of IOWA that also means all assets have been cached and the page works offline.
+- **supported**: the browser supports service worker, but the service worker is not yet controlling the page. This is the expected status for first time visitors.
+- **unsupported**: the user's browser does not support service worker.
+
+<pre class="prettyprint">
+function getServiceWorkerStatus() {
+  if ('serviceWorker' in navigator) {
+    return navigator.serviceWorker.controller ? 'controlled' : 'supported';
+  } else {
+    return 'unsupported';
+  }
+}
+</pre>
+
+This function got the service worker status for us; the next step was to associate this status with the data we were sending to Google Analytics.
+
+#### Tracking custom data with custom dimensions
+
+By default, Google Analytics gives you lots of ways to subdivide your total traffic into groups based on attributes of the user, session, or interaction. These attributes are known as [dimensions](https://support.google.com/analytics/answer/1033861). Common dimensions web developers care about are things like [Browser](/analytics/devguides/reporting/core/dimsmets#view=detail&group=platform_or_device&jump=ga_browser), [Operating System](/analytics/devguides/reporting/core/dimsmets#view=detail&group=platform_or_device&jump=ga_operatingsystem), or [Device Category](/analytics/devguides/reporting/core/dimsmets#view=detail&group=platform_or_device&jump=ga_devicecategory).
```

Comment:
What I'd been doing is capitalizing when using the actual dimension name; then using a link if mentioning it for the first time and italicizing on subsequent mentions. I wanted to make it clear that these weren't just generic terms, they were official terms and literal names. WDYT?